File: app/src/main/java/osama/me/viewmodelannotations/MainItemViewHolder.java
package osama.me.viewmodelannotations;

import android.view.View;

import osama.me.vmannotation.BindFields;

public class MainItemViewHolder extends BaseViewHolder<AKindOfObject> {

    private View itemView;

    MainItemViewHolder(final View itemView) {
        super(itemView);
        this.itemView = itemView;
    }

    @BindFields(viewIds = {"name", "amount"}, viewName = "itemView")
    @Override
    protected void bind(final AKindOfObject item) {
        super.bind(item);
        MainItemViewHolder_BindFields.bindfields(item, itemView);
    }
}

File: app/src/main/java/osama/me/viewmodelannotations/MainItemViewHolder.kt
package osama.me.viewmodelannotations

import android.view.View
import android.widget.TextView

//Proposed location for BindFields annotation to generate lines such as 11 and 12
class MainItemViewHolder(itemView : View) : BaseViewHolder<AKindOfObject>( itemView) {

    override fun bind(item: AKindOfObject?) {
        super.bind(item)
        itemView.findViewById<TextView>(R.id.name).text = item?.name
        itemView.findViewById<TextView>(R.id.amount).text = item?.amount.toString()
    }
}